### DISCUSSION TITLE
How to pass in a function with lightningcli configration?

### DISCUSSION THREAD

**Discussion by calvinzhan** [Lightning-AI/pytorch-lightning#14530] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Suppose I have a subclass of LightningDataModule

class TxtDataModule(LightningDataModule):
    def \__init\__(
        self,
        line_processor,  # line_processor is a function 
        ...
    ):
       line_processor(....)
       ....

How can I configure TxtDataModule to pass in a function in yaml configuration?

**Comment by tshu-w** [comment, marked as answer]
You may find [this](https://github.com/Lightning-AI/lightning/discussions/13613#discussioncomment-3174735) helpful.

**Reply by calvinzhan** [reply]
Thanks a lot, it works
